sw/device/silicon_creator/lib/shutdown_unittest.cc

```diff
@@ -0,0 +1,399 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/shutdown.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/testing/mask_rom_test.h"
+#include "sw/device/silicon_creator/lib/base/mock_abs_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/mock_alert.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
+
+// FIXME: I can't get ARRAYSIZE from `memory.h` because the definitions of
+// memcpy and friends there conflict with the standard definitions.
+#define ARRAYSIZE(x) ((sizeof x) / (sizeof x[0]))
+
+namespace shutdown_unittest {
+namespace {
+extern "C" {
+// Dummy out base_printf.
```

Comment:
I think we may want to think of a way to link `base_printf` into host binaries if possible...

Reply:
I'm all ears on ideas.

Should we just `#define base_printf printf` in `print.h` and wrap that in some `#ifdef` junk?  Or should we write a wrapper and pass a `va_list` off to `vfprintf` or somesuch?

Reply:
Honestly I would just use `base_printf.h` directly and add some code in `print.c` that detects whether we're on a host and, if so, defaults to using stdout or whatever. We can file and issue and deal with it later.